packages/main/src/DatePicker.js

```diff
@@ -476,6 +478,10 @@ class DatePicker extends DateComponentBase {
 			}
 		}
 
+		if ((this._getInput() === event.target && this.isOpen()) && (isTabNext(event) || isTabPrevious(event))) {
```

Comment:
Think its better to use this._getInput().isEqualNode(event.target) as it seems to be supported in all modern browsers.
And also this works for Tab and Shit + Tab combinations, but doesn't work for F6 and Shift + F6 group navigation combinations. The same works in ui5 classic.